dcos_test_utils/jobs.py

```diff
@@ -79,10 +92,18 @@ def _wait_for_run_completion(j_id: str, r_id: str) -> bool:
             # 404 means the run is complete and this is done
             # anything else is a problem and should not happen
             if rc.status_code == 404:
-                log.info('Job run {} finished.'.format(r_id))
-                return True
-            raise requests.HTTPError(
-                'Unexpected status code for job run {}: {}'.format(r_id, rc.status_code), response=rc)
+                history_available = self._is_history_available(j_id, r_id)
+                if history_available:
+                    log.info('Job run {} finished.'.format(r_id))
+                    return True
+                else:
+                    raise requests.HTTPError(
```

Comment:
LGTM.

But there isn't a test case for this scenario.

If we mock `_is_history_available` to False and our Exception raised can verified for 
`Waiting for job run {} to be finished, but history for that job run is not available'` in the message.